src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -1111,25 +1111,19 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     //We will use the client-supplied image, if present, otherwise we will use a default.
     //If welder is not enabled, we won't use any image.
     //Eventually welder will be enabled for all clusters and this will be way cleaner.
-    val welderImage = if(clusterRequest.enableWelder.getOrElse(false)) {
+    val welderImageOpt: Option[ClusterImage] = if(clusterRequest.enableWelder.getOrElse(false)) {
       val i = clusterRequest.welderDockerImage.getOrElse(dataprocConfig.welderDockerImage)
       Some(ClusterImage(Welder, i, now))
     } else None
 
-    val images = Set(
```

Comment:
This is the meat of the fix. If `enableWelder` is set to true, and the user doesn't provide a Jupyter image, this method will return a `Set(welderImage)` and not install Jupyter!